### PR TITLE
Customize arrows

### PR DESCRIPTION
**Pull request by ukscid**
here the patch for customize variation/comment arrows and last move arrow.

### CODE REVIEW

File: tcl/board.tcl [hunk 1]
```diff
@@ -1307,9 +1307,11 @@ proc ::board::mark::DrawArrow {pathName from to color} {
   if {$from < 0  ||  $from > 63} { return }
   if {$to   < 0  ||  $to   > 63} { return }
   set coord [GetArrowCoords $pathName $from $to]
+  set s1 [expr 3 * $::arrowWidth]
+  set shape [list $s1 $s1 [expr 1.5 * $::arrowWidth]]
   eval $pathName \
       {create line $coord} \
-      -fill $color -arrow last -width 2 \
+      -fill $color -arrow last -width $::arrowWidth {-arrowshape $shape} \
```
Comment (benini) [inline tcl/board.tcl]: Don't use global variables. Use parameters instead:
`proc ::board::mark::DrawArrow {pathName from to color width} { ...`

File: tcl/options.tcl [hunk 2]
```diff
@@ -213,6 +213,7 @@ set translatePieces 1
 
 # Highlight the last move played
 set arrowLastMove 0
+set arrowWidth 2
```
Comment (benini) [inline tcl/options.tcl]: arrowWidth is not related to the last move and should not be here:

```
# Default size for the board's arrows.
set boardArrowWidth 2

```

File: tcl/board.tcl [hunk 3]
```diff
@@ -1303,6 +1303,17 @@ proc ::board::mark::DrawText {pathName square char color {size 0} {shadowColor "
 
 # ::board::mark::DrawArrow --
 #
+proc ::board::mark::DrawArrowLastMove {pathName from to color} {
```
Comment (benini) [inline tcl/board.tcl]: Unnecessary

File: tcl/board.tcl [hunk 4]
```diff
@@ -1595,7 +1606,7 @@ proc  ::board::lastMoveHighlight {w moveuci} {
         ::board::mark::DrawRectangle $w.bd $square2 $::highlightLastMoveColor $::highlightLastMovePattern
     }
     if { $::arrowLastMove } {
-        ::board::mark::DrawArrow $w.bd $square1 $square2 $::highlightLastMoveColor
+        ::board::mark::DrawArrowLastMove $w.bd $square1 $square2 $::highlightLastMoveColor
```
Comment (benini) [inline tcl/board.tcl]: `::board::mark::DrawArrow $w.bd $square1 $square2 $::highlightLastMoveColor $::highlightLastMoveArrowWidth`

File: tcl/options.tcl [hunk 5]
```diff
@@ -214,6 +214,7 @@ set translatePieces 1
 # Highlight the last move played
 set arrowLastMove 0
 set arrowWidth 2
+set highlightArrowWidth 2
```
Comment (benini) [inline tcl/options.tcl]: use consistent and clear names for variables, expecially for global variables:
```
set lastMoveShowArrow 0
set lastMoveArrowWidth 2
```

File: tcl/windows/preferences.tcl [hunk 6]
```diff
@@ -177,6 +177,10 @@ proc ::preferences::moves { t } {
     ttk::checkbutton $t.sva -variable showVarArrows -text [tr OptionsMovesShowVarArrows]
     ttk::checkbutton $t.god -variable glossOfDanger -text [tr OptionsMovesGlossOfDanger] -command updateBoard
 
+    ttk::frame $t.arrow
+    ttk::label $t.arrow.label -text "Width of Arrows for\nVariations and comments"
```
Comment (benini) [inline tcl/windows/preferences.tcl]: The expression "Arrows for comments" is meaningless.

**Review by benini — COMMENTED** [review]
It took me some time to highlight all these things in details. And I did that because, quite honestly, they should be obvious.

**Comment by ukscid** [comment]
Variable names changed.

changed  ::board::mark::DrawArrowLastMove to call with width parameter.

not yet changed: ::board::mark::DrawArrow . This is called via
    if {[catch {eval $drawingScript $board $square $dest $color}]} {
should this changed like
    proc ::board::mark::DrawArrow {pathName from to color {width $::boardArrowWidth} }  ?

The text "Width of Arrows for\nVariations and Comments" should
indicate, that the width is used for Variants arrows and the arrows
which are drawn when in a comment [%draw arrow ..] is used.
Other suggestions for the wording?

**Comment by ukscid** [comment]
now remove ::DrawArrowLastMove

**Pull request closed without merging**